### DISCUSSION TITLE
Estimation of Standard Deviation of Shocks

### DISCUSSION THREAD

**Discussion by tosapola** [IRIS-Solutions-Team/IRIS-Toolbox#357] [General]
Hi,

The question may sound simple, but I'm confused about this for years.
What is the best practice for estimation std dev of shocks?
Since normally use relative std dev in Kalman Filter, so, is it correct that we always have to keep std dev of a shock fixed as "numeraire"? If yes, how should we report "true" std dev of all shocks? And yes, there is a scaling factor, but should it apply to all of those Bayesian elements, i.e., prior, posterior, mode.

Or, there is another way to do it properly by turning relative off? And this way, finding the appropriate starting points seems to be a challenge.

Pros/Cons of both ways?

Best,
Tos

**Comment by jaromir-benes** [comment]
Hi Tos - honestly, this has been confusing for me myself as well :)) For a
long time, I believed that relative=true is the way to go, keeping one of
the std devs as the "numeraire", and estimating the common variance/std
scale factor in each run of the kalman filter.

But from my hands-on practical experience, it does always work that way
(remains a bit of mystery to me, still:).

So in short, I don't know :)

Also notice that because of the confusion the relative=true option
sometimes introduces, I chose to declare the default value for the option
to be false in the new implementation of the kalman filter - the
"kalmanFilter" function, which is also used in the estimate function.

Perhaps a disappointing answer, but that's life...

Best
Jaromir

**Reply by tosapola** [reply]
Thank you so much. At least not me alone confused by this :)

Oh, didn't notice that you turn it off in the new kalmanFilter(). Please note that the doc still says Relative='true' at default.

Best,
Tos

**Comment by whogeeyao** [comment]
Hi, 

Would the `Override=` option in `kalmanFilter` re-estimate the standard deviations of shocks, or do I have to supply the SD myself?

Currently I estimated my parameters of a model using the estimate function, and want to update my SD of shocks before running the Kalman Filter.
